File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] cnt = new int[26];
        for(int i=0; i<magazine.length(); i++) {
            cnt[magazine.charAt(i)-'a']++;
        }

        for(int i=0; i<ransomNote.length(); i++) {
            char c = ransomNote.charAt(i);
            cnt[c-'a']--;
            if(cnt[c-'a']<0) return false;
        }
        return true;
    }
}

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        arr = [0] * 26

        for char in magazine:
            arr[ord(char) - ord('a')] += 1
        
        for char in ransomNote:
            if arr[ord(char) - ord('a')] == 0:
                return False
            arr[ord(char) - ord('a')] -= 1
        return True
